lib/Basics/ResultT.h

```diff
@@ -152,6 +154,15 @@ class ResultT {
   // cppcheck-suppress noExplicitConstructor
   /* implicit */ ResultT(T const& val) : ResultT(val, TRI_ERROR_NO_ERROR) {}
 
+  // Default constructor calls default constructor of T
+  // If T is not default constructable, ResultT cannot be default constructable
+  template<typename U = T,
+           std::enable_if_t<std::is_nothrow_default_constructible<U>::value,
+                            bool> = true>
+  ResultT() : _val{T{}} {}
+  template<typename U = T,
+           std::enable_if_t<!std::is_nothrow_default_constructible<U>::value,
+                            bool> = true>
   ResultT() = delete;
```

Comment:
Are you sure its a good idea to make such a change halfway?

I would opt to first commit this as is, and then introduce concepts for inspectors as a separate PR.

Reply:
That was just a suggestion to make the code a bit more readable, but feel free to ignore it! :)

Reply:
Disregard my comment, I overlooked that this is code in `ResultT`, not in inspectors.

Since concepts also give us hope of better error messages, lets try using them here and see what happens in CI